### PR TITLE
fix(concourse): support re-run (dot-) builds

### PR DESCRIPTION
**Pull request by jaredstehler**
Concourse 6.x introduces "re-run" builds, which have a decimal build number (i.e. 12.1) breaking the existing getNumber() method expecting an int. This change will allow igor to properly parse these build numbers, and will collapse the build listing wherein each number is the max(decimalNumber).

Addresses issue: https://github.com/spinnaker/spinnaker/issues/5773

### CODE REVIEW

File: igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2016 Schibsted ASA.
```
Comment (jervi) [inline igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy:2]: While I'd love for my organisation to take credit for your test, you might want to update this 😉

File: igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+{"data":{"origin":{"id":"5ed14571"},"time":1591654963,"exit_status":0,"version":{"ref":"65a7c129fe402634ee1d94b5ba2577fd46a039f2"},"metadata":[{"name":"commit","value":"65a7c129fe402634ee1d94b5ba2577fd46a039f2"},{"name":"author","value":"John Smith"},{"name":"author_date","value":"2020-06-08 15:13:23 -0700"},{"name":"branch","value":"master"},{"name":"tags","value":"v1.0.178"},{"name":"message","value":"DOOL-1990: reopening question sets completionPercent to zero\\n\\n"},{"name":"url","value":"https://github.com/myteam/gateway/commit/65a7c129fe402634ee1d94b5ba2577fd46a039f2"}]},"event":"finish-get","version":"5.1"}
+        ]'''
```
Comment (jervi) [inline igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy:329]: Nit: Consider indenting the lines and rather add a `.stripIndent()` after the ending triple quotes. Same with the json in the `planFixture()` method as well.

**Comment by gal-yardeni** [comment]
ping @jaredstehler can you please follow up per @jervi review? thanks!

File: igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy (hunk 1 above)
Comment (jaredstehler) [reply igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy:2]: done

File: igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy (hunk 2 above)
Comment (jaredstehler) [reply igor-web/src/test/groovy/com/netflix/spinnaker/igor/concourse/service/ConcourseServiceSpec.groovy:329]: done

**Comment by jaredstehler** [comment]
> ping @jaredstehler can you please follow up per @jervi review? thanks!

all set!

File: igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java [hunk 3]
```diff
@@ -327,14 +339,21 @@ public JobConfiguration getJobConfig(String jobName) {
...
             job.getPipelineName(),
             job.getName(),
             host.getBuildLookbackLimit(),
-            since);
+            since)
+        .stream()
+        .sorted()
+        .collect(
+            Collectors.toMap(
+                (b) -> b.getNumber(), Function.identity(), (b1, b2) -> b1, LinkedHashMap::new))
```
Comment (jervi) [inline igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java:353]: One more thing: Is there a side effect here that I’m overlooking? Looks like this entire collector + the next call to values() and stream() should just be removed?

Comment (jaredstehler) [reply igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java:353]: I'm sure there's a prettier way to express this, but I am collapsing the list on the latest build for each major build number. e.g.: `14, 13.1, 13, 12` would collapse into `14, 13.1, 12`

Comment (jaredstehler) [reply igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java:353]: (see where I've updated `Build` to be `Comparable` on `decimalNumber`)

Comment (jervi) [reply igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/service/ConcourseService.java:353]: Gotcha, thanks!

**Comment by gal-yardeni** [comment]
hi @jaredstehler can you rebase the branch? not sure why it didn't get auto-merged.

**Comment by jaredstehler** [comment]
> hi @jaredstehler can you rebase the branch? not sure why it didn't get auto-merged.

are there specific instructions for that beyond clicking the inline button to update?

**Comment by gal-yardeni** [comment]
@jaredstehler nope..  I think I needed to add my review as well for the auto-merge to merge it. Anyway, it got merged now! thank you for being so responsive!

**Comment by jaredstehler** [comment]
np, happy to help!

**Pull request merged**